source/val/validate_image.cpp

```diff
@@ -1457,12 +1457,22 @@ spv_result_t ValidateImageGather(ValidationState_t& _,
   }
 
   if (opcode == SpvOpImageGather || opcode == SpvOpImageSparseGather) {
-    const uint32_t component_index_type = _.GetOperandTypeId(inst, 4);
+    const uint32_t component = inst->GetOperandAs<uint32_t>(4);
+    const uint32_t component_index_type = _.GetTypeId(component);
     if (!_.IsIntScalarType(component_index_type) ||
         _.GetBitWidth(component_index_type) != 32) {
       return _.diag(SPV_ERROR_INVALID_DATA, inst)
              << "Expected Component to be 32-bit int scalar";
     }
+    if (spvIsVulkanEnv(_.context()->target_env)) {
+      if (!spvOpcodeIsConstant(_.GetIdOpcode(component)) &&
```

Comment:
This covers spec constants too.

Reply:
The Vulkan spec says 

> must be the <id> of a constant instruction

So does that not suppose to include spec constants as well?

Reply:
It does. `spvOpcodeIsConstant` checks for spec constants too I meant. You can drop the specific spec constant check.

Reply:
... 🤦 I didn't see the `spvOpcodeIsSpecConstant` I still left, yes will definitely remove, thanks for catching